graphdatascience/model/simple_rel_embedding_model.py

```diff
@@ -0,0 +1,246 @@
+import os
+from typing import Any, Dict, List, Union
+
+from pandas import DataFrame, Series
+
+from ..query_runner.query_runner import QueryRunner
+from ..server_version.server_version import ServerVersion
+
+NodeFilter = Union[int, List[int], str]
+
+
+class SimpleRelEmbeddingModel:
+    """
+    A class whose instances represent a model for computing and ranking pairwise distance between nodes
+    according to knowledge graph style metrics. It may also produce new relationships based on these
+    rankings.
+    """
+
+    def __init__(
+        self,
+        scoring_function: str,
+        query_runner: QueryRunner,
+        server_version: ServerVersion,
+        graph_name: str,
+        node_embedding_property: str,
+        relationship_type_embeddings: Dict[str, List[float]],
+    ):
+        self._scoring_function = scoring_function
+        self._query_runner = query_runner
+        self._server_version = server_version
+        self._graph_name = graph_name
+        self._node_embedding_property = node_embedding_property
+        self._relationship_type_embeddings = relationship_type_embeddings
+
+    def predict_stream(
```

Comment:
I think we should also put compatible_with min_version=2.5.0 for these 3 predict methods?

Reply:
Or, is it because compatible_with is on the create method for DistMultCreator and TransECreator, they are no longer needed here?

Reply:
Somehow tests failing on gds.model.transe.create with parameters ['G', 'node_embedding_property', 'relationship_type_embeddings'] requires GDS server version >= 2.5.0. The current version is 2.4.7.

But it has a @compatible_with , and shouldn't have been run on 2.4?

Reply:
Yeah I think it's fine since they cannot create the models at all on < 2.5. Looks like it's the doc tests which are failing which makes sense, so will look into that